kotlinpoet/src/main/java/com/squareup/kotlinpoet/FunSpec.kt

```diff
@@ -359,6 +368,23 @@ public class FunSpec private constructor(
       typeVariables += typeVariable
     }
 
+    @DelicateKotlinPoetApi("Context receivers are currently a preview feature")
+    public fun contextReceiver(receiverTypes: Collection<TypeName>): Builder = apply {
+      check(!name.isConstructor) { "$name cannot have receiver type" }
+      contextReceiverTypes += receiverTypes
+    }
+
+    @DelicateKotlinPoetApi("Context receivers are currently a preview feature")
+    public fun contextReceiver(vararg receiverType: TypeName): Builder = contextReceiver(receiverType.toList())
+
+    @DelicateKotlinPoetApi("Context receivers are currently a preview feature")
+    public fun contextReceiver(vararg receiverType: Type): Builder =
```

Comment:
Remove this and the KClass overload below this and _only_ expose TypeName overloads. I think you misunderstood my previous response. We should have two - a vararg param and an Iterable plural overload